gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -416,6 +420,31 @@ public final Strictness getStrictness() {
     return strictness;
   }
 
+  /**
+   * Sets whether multiple top-level values are allowed. Only whitespace is supported as separator
+   * between top-level values. Values may also be concatenated without any whitespace in between,
+   * but for some values this causes ambiguities, for example for JSON numbers as top-level values.
+   *
+   * <p>This setting overwrites and is overwritten by whether {@link #setStrictness(Strictness)}
+   * enabled support for multiple top-level values.
+   *
+   * @see #isMultiTopLevelValuesAllowed()
+   * @since $next-version$
+   */
+  public final void setMultiTopLevelValuesAllowed(boolean enabled) {
+    this.multiTopLevelValuesEnabled = enabled;
+  }
+
+  /**
+   * Returns whether multiple top-level values are allowed.
+   *
+   * @see #setMultiTopLevelValuesAllowed(boolean)
+   * @since $next-version$
+   */
+  public final boolean isMultiTopLevelValuesAllowed() {
+    return multiTopLevelValuesEnabled;
+  }
```

Comment:
I am not completely happy about these method names `setMultiTopLevelValuesAllowed` and `isMultiTopLevelValuesAllowed` yet. And feedback and suggestions are welcome.